pkg/storegateway/indexheader/reader_pool.go

```diff
@@ -69,20 +98,41 @@ func NewReaderPool(logger log.Logger, lazyReaderEnabled bool, lazyReaderIdleTime
 				}
 			}
 		}()
+
+		go func() {
+			for {
+				select {
+				case <-p.close:
+					return
+				// Persist lazy loaded blocks in file every one minute
+				case <-time.After(1 * time.Minute):
+					// We copy the state of blocks that are being lazy loaded from LazyBinaryReaders.
+					p.lazyReadersMx.Lock()
+					p.lazyLoadedTracker.CopyLazyLoadedState(p.lazyReaders)
+					p.lazyReadersMx.Unlock()
+
+					// Then we persist the state to files.
+					if err := p.lazyLoadedTracker.Persist(); err != nil {
+						level.Warn(p.logger).Log("msg", "failed to persist lazyLoadedTracker", "err", err)
```

Comment:
Can you clarify this message? I'm thinking from the point of view of an operator, they don't know what `lazyLoadedTracker` is. 
```suggestion
						level.Warn(p.logger).Log("msg", "failed to persist list of lazy-loaded index headers", "err", err)
```